src/registry_testing.jl

```diff
@@ -15,6 +15,33 @@ else
 end
 
 is_valid_url(str::AbstractString) = !isempty(HTTP.URI(str).scheme) && isvalid(HTTP.URI(str))
+
+function _include_this_registry(registry_spec, registry_deps_names::Vector{<:AbstractString})
+    all_fieldvalues = []
+    for fieldname in [:name, :repo, :url]
+        if hasproperty(registry_spec, fieldname)
+            fieldvalue = getproperty(registry_spec, fieldname)
+            if fieldvalue isa AbstractString
+                push!(all_fieldvalues, fieldvalue)
+            end
+        end
+    end
+    for fieldvalue in all_fieldvalues
+        for registry_deps_name in registry_deps_names
+            if strip(fieldvalue) == strip(registry_deps_name)
+                return true
+            end
+            if strip(fieldvalue) == strip("$(registry_deps_name).git")
+                return true
+            end
+            if strip("$(fieldvalue).git") == strip(registry_deps_name)
```

Comment:
So the bug was caused because I did not include the full link with `.git` at the end?

Reply:
Nope. The bug was due to a change in the Pkg internals. Because we didn't have any unit tests covering this code, we did not detect the change.

I took this opportunity to also add some code that takes care of links both with and without `.git` at the end. But the main bug was unrelated to that.

Reply:
Thanks